File: GIFrameworkMaps.Web/Scripts/Interfaces/OGCMetadata/LayerResource.ts
﻿import { FeaturePropertyType } from "@camptocamp/ogc-client";
import { Extent } from "ol/extent";

export interface LayerResource {
  name: string;
  title: string;
  abstract: string;
  attribution: string;
  formats: string[];
  baseUrl: string;
  projections: string[];
  extent: Extent;
  queryable: boolean;
  opaque: boolean,
  version: string;
  proxyMetaRequests: boolean;
  proxyMapRequests: boolean;
  keywords?: string[];
  properties: Record<string,FeaturePropertyType>
}

File: GIFrameworkMaps.Data/Models/LayerResource.cs
using System.Text.Json.Serialization;

namespace GIFrameworkMaps.Data.Models
{
	public class LayerResource
	{
		[JsonPropertyName("name")]
		public string? Name { get; set; }
		[JsonPropertyName("title")]
		public string? Title { get; set; }
		[JsonPropertyName("abstract")]
		public string? Abstract { get; set; }
		[JsonPropertyName("attribution")]
		public string? Attribution { get; set; }
		[JsonPropertyName("formats")]
		public string[] Formats { get; set; } = [];
		[JsonPropertyName("baseUrl")]
		public string? BaseUrl { get; set; }
		[JsonPropertyName("projection")]
		public string[] Projections { get; set; } = [];
		[JsonPropertyName("extent")]
		public string[] Extent { get; set; } = [];
		[JsonPropertyName("version")]
		public string? Version { get; set; }
		[JsonPropertyName("proxyMetaRequests")]
		public bool ProxyMetaRequests { get; set; }
		[JsonPropertyName("proxyMapRequests")]
		public bool ProxyMapRequests { get; set; }
	}
}
